client-js/app/scripts/app.ts

```diff
@@ -364,28 +413,39 @@ window.addEventListener('WebComponentsReady', () => {
    * If the login fails due to an invalid password, we clear the credentials
    * from this storage object.
    */
-  async function attemptLogin(mgr: manager.ManagerApi, manager: string, storage: Storage): Promise<void> {
+  async function attemptLogin(mgr: ManagerApi, manager: string, storage: Storage): Promise<void> {
     var key = 'labrad-credentials';
     if (manager) {
       key += '.' + manager;
     }
-    var creds = loadCreds(manager, storage);
-    if (creds === null) {
+    var credential = loadCredential(manager, storage);
+    if (credential === null) {
       throw new Error('no credentials');
-    } else {
-      try {
-        await mgr.login({
-          username: creds.username,
-          password: creds.password,
-          host: manager
-        });
-      } catch (error) {
-        if (isPasswordError(error)) {
-          // if we had credentials, clear them out
-          storage.removeItem(key);
-        }
-        throw error;
+    }
+    try {
+      // TODO: in TS2.0 compiler will understand .kind; can then remove casts
```

Comment:
Full stop needed. Also, it's recommended that you do "TODO (username)" so that future people know who to contact regarding the TODO, it's not considered signing up to do the change.